test/refactoring/ConvertAsync/convert_async_attributed_renames.swift

```diff
@@ -0,0 +1,95 @@
+  }
+  // and me
+  nonCompletionName { str in
+    print(str)
+  }
+}
+// SIMPLERENAMED: func simpleRenamed() async {
+// SIMPLERENAMED-NEXT: // preserve me
+// SIMPLERENAMED-NEXT: let str = await simple2()
+// SIMPLERENAMED-NEXT: print(str)
+// SIMPLERENAMED-NEXT: // and me
+// SIMPLERENAMED-NEXT: let str1 = await simple2()
+// SIMPLERENAMED-NEXT: print(str1)
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=RENAMEDCOLLISION %s
+func renamedCollision() {
+  simple { simple2 in
+    print(simple2)
+  }
+}
+// RENAMEDCOLLISION: func renamedCollision() async {
+// RENAMEDCOLLISION-NEXT: let simple21 = await simple2()
+// RENAMEDCOLLISION-NEXT: print(simple21)
+
+@available(*, renamed: "simpleArg2")
+func simpleArgRenamed(arg: String, _ random: @escaping (String) -> Void) { }
+@available(*, renamed: "simpleArg2")
+func completionFirstArg(random: @escaping (String) -> Void, arg: String) { }
+func simpleArg2(newArg: String) async -> String { }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=ARGRENAMED %s
+func argRenamed() {
+  (simpleArgRenamed)(arg: "foo") { str in
+    print(str)
+  }
+  (completionFirstArg)(random: { str in
+    print(str)
+  }, arg: "foo")
+}
+// ARGRENAMED: func argRenamed() async {
+// ARGRENAMED-NEXT: let str = await (simpleArg2)(newArg: "foo")
+// ARGRENAMED-NEXT: print(str)
+// ARGRENAMED-NEXT: let str1 = await (simpleArg2)(newArg: "foo")
+// ARGRENAMED-NEXT: print(str1)
+
+@available(*, renamed: "multiHandlers2")
+func multiHandlers(arg: String, handler1: @escaping (String) -> Void, handler2: @escaping (String) -> Void) { }
+func multiHandlers2(newArg: String, newHandler: @escaping (String) -> Void) async -> String { }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=MULTIPLEHANDLERS %s
+func multipleHandlers() {
+  multiHandlers(arg: "foo", handler1: { str1 in
+    print(str1)
+  }, handler2: { str2 in
+    print(str2)
+  })
+  multiHandlers(arg: "foo", handler1: { str3 in
+    print(str3)
+  }) { str4 in
+    print(str4)
+  }
+  multiHandlers(arg: "foo") { str5 in
+    print(str5)
+  } handler2: { str6 in
+    print(str6)
+  }
+}
+// MULTIPLEHANDLERS: func multipleHandlers() async {
+// MULTIPLEHANDLERS-NEXT: let str2 = await multiHandlers2(newArg: "foo", newHandler: { str1 in
+// MULTIPLEHANDLERS-NEXT: print(str1)
+// MULTIPLEHANDLERS-NEXT: })
+// MULTIPLEHANDLERS-NEXT: print(str2)
+// MULTIPLEHANDLERS-NEXT: let str4 = await multiHandlers2(newArg: "foo", newHandler: { str3 in
+// MULTIPLEHANDLERS-NEXT: print(str3)
+// MULTIPLEHANDLERS-NEXT: })
+// MULTIPLEHANDLERS-NEXT: print(str4)
+// MULTIPLEHANDLERS-NEXT: let str6 = await multiHandlers2(newArg: "foo", newHandler: { str5 in
+// MULTIPLEHANDLERS-NEXT: print(str5)
+// MULTIPLEHANDLERS-NEXT: })
+// MULTIPLEHANDLERS-NEXT: print(str6)
```

Comment:
Some suggestions for more test cases:

- The renamed method doesn’t exist
- The renamed method has more parameters than the original once
- The renamed method has fewer parameters than the original one
- The renamed method has more parameters than the original one but all of the added parameters are defaulted (either at the beginning/middle and at the end)
- The renamed method has changed the parameter order (and kept the labels)
- The renamed method still has the completion parameter and only changed its base name / labels

I don’t think we need to produce perfect results in all of those cases, but I think it would be good to document what we do for them.

Reply:
So my preference was to test the attribute -> decl matching in `attr_availability_async_rename` rather than here, though I'm probably missing some of the tests you mentioned there.

The answer to all of them is that if a renamed decl isn't found, it isn't used. This would fallback to the default behaviour of assuming one exists with the same name/params. Would you be okay if I just added the single "the renamed method doesn't exist" and make sure to test those others in `attr_availablity_async_rename`?

Reply:
So we consider the async alternative as not found even if it exists but the number of arguments doesn’t match?

E.g. in
```swift
func foo(x: Int) {}

@available(*, renamed: "foo(x:)")
func bar(x: Int, y: Int, completion: @escaping (Int) -> Void) {}
```

`bar`’s async alternative is `bar(x:y:)` because we are ignoring the attribute since the argument count doesn’t match up?

I think the defaulted argument case would be worth adding as I can see this happening as the API of the async alternative evolves. 

I also don’t really care where these test cases go, it was just a bunch of ideas that I though would be worth testing in general.

Reply:
Yep, arguments have to match up (including types) minus the completion handler.